exporter/azuremonitorexporter/traceexporter.go

```diff
@@ -0,0 +1,535 @@
+func fillRemoteDependencyDataInternal(span *tracepb.Span, data *contracts.RemoteDependencyData) {
+	data.Type = "InProc"
+
+	// Everything is a custom attribute here
+	for k, v := range span.Attributes.AttributeMap {
+		data.Properties[k] = attributeValueAsString(v)
+	}
+}
+
+// Sets properties on an AppInsights RemoteDependencyData from the wire format Span when the outbound Span type is HTTP
+func fillRemoteDependencyDataHTTP(span *tracepb.Span, data *contracts.RemoteDependencyData) {
+	data.Type = "Http"
+	data.ResultCode = "200"
+
+	/*
+		The set of expected OpenTelemetry attribute sets for HTTP client span:
+		https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-http.md#http-client
+
+		Order of preference is:
+			http.url
+			http.scheme, http.host, http.target
+			http.scheme, peer.hostname, peer.port, http.target
+			http.scheme, peer.ip, peer.port, http.target
+	*/
+	httpMethod := ""
+	httpURL := ""
+	httpScheme := ""
+	httpHost := ""
+	httpTarget := ""
+	peerHostName := ""
+	peerPort := ""
+	peerIP := ""
+
+	for k, v := range span.Attributes.AttributeMap {
+		stringValue := attributeValueAsString(v)
+		switch k {
+		case spanAttributeKeyComponent:
+			// do nothing
+		case spanAttributeKeyHTTPMethod:
+			httpMethod = stringValue
+		case spanAttributeKeyHTTPUrl:
+			httpURL = stringValue
+		case spanAttributeKeyHTTPScheme:
+			httpScheme = stringValue
+		case spanAttributeKeyHTTPHost:
+			httpHost = stringValue
+		case spanAttributeKeyHTTPTarget:
+			httpTarget = prefixIfNecessary(stringValue, "/")
+		case spanAttributeKeyPeerHostname:
+			peerHostName = stringValue
+		case spanAttributeKeyPeerIP:
+			peerIP = stringValue
+		case spanAttributeKeyPeerPort:
+			peerPort = stringValue
+		case spanAttributeKeyHTTPStatusCode:
+			data.ResultCode = stringValue
+			code, err := strconv.Atoi(data.ResultCode)
+			if err == nil {
+				data.Success = code >= 200 && code <= 399
+			}
+		default:
+			// Anything not on this list above ends up as a custom property
+			data.Properties[k] = stringValue
+		}
+	}
+
+	if httpURL != "" {
+		if u, err := url.Parse(httpURL); err == nil {
+			data.Name = httpMethod + " " + u.Path
+			data.Data = httpURL
+			data.Target = u.Host
+		}
+	} else if httpScheme != "" && httpHost != "" && httpTarget != "" {
+		data.Name = httpMethod + " " + httpTarget
+		data.Data = httpScheme + "://" + httpHost + httpTarget
+		data.Target = httpHost
+	} else if httpScheme != "" && peerHostName != "" && peerPort != "" && httpTarget != "" {
+		data.Name = httpMethod + " " + httpTarget
+		data.Data = httpScheme + "://" + peerHostName + ":" + peerPort + httpTarget
+		data.Target = peerHostName + ":" + peerPort
+	} else if httpScheme != "" && peerIP != "" && peerPort != "" && httpTarget != "" {
+		data.Name = httpMethod + " " + httpTarget
+		data.Data = httpScheme + "://" + peerIP + ":" + peerPort + httpTarget
+		data.Target = peerIP + ":" + peerPort
+	}
```

Comment:
if these are gonna be called a lot  on the hot path, may be string builder would be better than concat?

Reply:
Will do.